csrc/scheduler/matmul.cpp

```diff
@@ -122,12 +123,17 @@ inline void checkConcreteStaticDim(IterDomain* id) {
 //! If the input tensorview is not stored in shared memory, the function will
 //! skip the actual swizzle. This is used to help the domain mapping between
 //! mma_result and the epilogue tensor.
-void swizzleSharedMemory(TensorView* shared_mem_tv) {
+//! Returns the domain with swizzle. For the case of legacy swizzle, this
+//! domain must be set as loop domain. For the case of new swizzle, this domain
+//! must be set as allocation domain.
+template <bool legacy = true>
```

Comment:
This must be a template to allow incremental development. In my next step, I will use `legacy = false` for epilogue, and `legacy = true` for prologue. The reason for not using `legacy = false` for prologue is because ldmatrix is extremely difficult to index, so I don't want to touch it yet.

Reply:
Makes sense!